cmd/notation/cert.go

```diff
@@ -11,94 +11,122 @@ import (
 	"github.com/notaryproject/notation/internal/ioutil"
 	"github.com/notaryproject/notation/internal/slices"
 	"github.com/notaryproject/notation/pkg/config"
-	"github.com/urfave/cli/v2"
+	"github.com/spf13/cobra"
 )
 
-var (
-	certCommand = &cli.Command{
-		Name:    "certificate",
+type certAddOpts struct {
+	path string
+	name string
+}
+
+type certRemoveOpts struct {
+	names []string
+}
+
+type certGenerateTestOpts struct {
+	name      string
+	bits      int
+	expiry    time.Duration
+	trust     bool
+	hosts     []string
+	isDefault bool
+}
+
+func certCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "certificate",
 		Aliases: []string{"cert"},
-		Usage:   "Manage certificates used for verification",
-		Subcommands: []*cli.Command{
-			certAddCommand,
-			certListCommand,
-			certRemoveCommand,
-			certGenerateTestCommand,
-		},
+		Short:   "Manage certificates used for verification",
 	}
 
-	certAddCommand = &cli.Command{
-		Name:      "add",
-		Usage:     "Add certificate to verification list",
-		ArgsUsage: "<path>",
-		Flags: []cli.Flag{
-			&cli.StringFlag{
-				Name:    "name",
-				Aliases: []string{"n"},
-				Usage:   "certificate name",
-			},
+	command.AddCommand(certAddCommand(nil), certListCommand(), certRemoveCommand(nil), certGenerateTestCommand(nil))
+	return command
+}
+
+func certAddCommand(opts *certAddOpts) *cobra.Command {
+	if opts == nil {
+		opts = &certAddOpts{}
+	}
+	command := &cobra.Command{
+		Use:   "add [path]",
+		Short: "Add certificate to verification list",
+		Args:  cobra.ExactArgs(1),
```

Comment:
If the `path` is an empty string, the command will still fail and the help message shows that it needs a `path` as an argument.
```
path := ctx.Args().First()
if path == "" {
    return errors.New("missing certificate path")
}
```

Reply:
> If the `path` is an empty string, the command will still fail and the help message shows that it needs a `path` as an argument.
> 
> ```
> path := ctx.Args().First()
> if path == "" {
>     return errors.New("missing certificate path")
> }
> ```

Yes, but if user inputs more than 1 args, cobra.ExactArgs(1) would fail but the orignal logic won't.

Reply:
I will use `Args` field to check args.